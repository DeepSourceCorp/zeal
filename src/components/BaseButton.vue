<template>
  <button class="bg-juniper text-ink-400 leading-8 font-normal py-2 px-4 rounded-sm">
    Default button
  </button>
</template>

<script>
export default {
  name: 'button',

  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>
