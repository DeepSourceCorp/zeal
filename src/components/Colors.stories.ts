import '../assets/css/tailwind.css'
import '../assets/css/typography.css'
import '../assets/css/layout.css'

import baseColors from '@/helpers/tailwind/colors'

export default {
  title: 'Colors'
}

export const Colors = () => ({
  data() {
    return {
      baseColors: baseColors
    }
  },
  template: `<div class="p-5">
      <h1 class="text-lg font-bold text-vanilla-100 mb-5">Pallette</h1>
      <div class='grid grid-cols-4 gap-5 mb-12'>
        <template v-for="(color, key) in baseColors">
          <div :key="key" v-if="typeof color !== 'string'">
            <div class="rounded-lg overflow-hidden shadow-lg">
              <template v-for="(shade, name) in color">
                <div
                  v-if="name !== 'DEFAULT'"
                  :key="name"
                  :class="'bg-' + key + '-' + name"
                  class="w-full h-12 flex items-center justify-between font-semibold px-4"
                >
                  <span>{{ name }}</span>
                  <span class="uppercase">{{ shade }}</span>
                </div>
              </template>
            </div>
            <div class="rounded-lg overflow-hidden shadow-lg mt-2" v-if="color['DEFAULT']">
                <div
                  :class="'bg-' + key"
                  class="w-full h-12 flex items-center justify-between font-semibold px-4"
                >
                  <span>Default</span>
                  <span class="uppercase">{{ color['DEFAULT'] }}</span>
                </div>
            </div>
            <h3 class="mt-4 text-base font-bold text-vanilla-100 capitalize">{{ key }}</h3>
            <pre class="text-vanilla-400 text-xs">{{ value }}</pre>
          </div>
        </template>
      </div>
      <h1 class="text-lg font-bold text-vanilla-100 mb-5">Single Colors</h1>
      <div class='grid grid-cols-8 gap-5 mb-8'>
        <template v-for="(value, key) in baseColors">
          <div :key="key" v-if="typeof value === 'string' && !['transparent', 'current'].includes(key)">
            <div class="w-full h-22 rounded-lg" :class="'bg-' + key"></div>
            <h3 class="mt-2 text-base font-bold text-vanilla-100 capitalize">{{ key }}</h3>
            <pre class="text-vanilla-400 text-xs">{{ value }}</pre>
          </div>
        </template>
      </div>
    </div>`
})
