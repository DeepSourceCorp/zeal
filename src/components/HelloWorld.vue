<template>
  <div class="hello">
    <h1>Hello World</h1>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

@Component
export default class HelloWorld extends Vue {}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
