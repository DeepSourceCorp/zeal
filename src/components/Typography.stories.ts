import '../assets/css/tailwind.css'
import '../assets/css/typography.css'
import '../assets/css/layout.css'

const text = `<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. <a href="http://loripsum.net/" target="_blank">Aufert enim sensus actionemque tollit omnem.</a> Sed quid sentiat, non videtis. Hic ambiguo ludimur. Duo Reges: constructio interrete. Si id dicis, vicimus. </p>

<ol>
	<li>Quam ob rem tandem, inquit, non satisfacit?</li>
	<li>Consequens enim est et post oritur, ut dixi.</li>
	<li>Idem iste, inquam, de voluptate quid sentit?</li>
</ol>


<pre>E quo efficitur, non ut nos non intellegamus quae vis sit
istius verbi, sed ut ille suo more loquatur, nostrum
neglegat.

Vives, inquit Aristo, magnifice atque praeclare, quod erit
cumque visum ages, numquam angere, numquam cupies, numquam
timebis.
</pre>

<h2>Heading 2</h2>

<p>Hoc non est positum in nostra actione. Mihi enim satis est, ipsis non satis. Quae diligentissime contra Aristonem dicuntur a Chryippo. Nos cum te, M. Illa videamus, quae a te de amicitia dicta sunt. </p>

<ul>
	<li>Nam illud vehementer repugnat, eundem beatum esse et multis malis oppressum.</li>
	<li>Sic enim maiores nostri labores non fugiendos tristissimo tamen verbo aerumnas etiam in deo nominaverunt.</li>
	<li>Nam Metrodorum non puto ipsum professum, sed, cum appellaretur ab Epicuro, repudiare tantum beneficium noluisse;</li>
	<li>Ab his oratores, ab his imperatores ac rerum publicarum principes extiterunt.</li>
</ul>

<h2>Heading 3</h2>

<p>Iam contemni non poteris. Quid enim est a Chrysippo praetermissum in Stoicis? Refert tamen, quo modo. Eadem fortitudinis ratio reperietur. Sint modo partes vitae beatae. Qua tu etiam inprudens utebare non numquam. Quid nunc honeste dicit? Beatus sibi videtur esse moriens. </p>`

export default {
  title: 'Typography'
}

export const DEFAULT = () => ({
  template: `<div class="grid grid-cols-2">
    <div class="p-5 prose">
      <h1>Default</h1>
      ${text}
    </div>
    <div class="p-5 prose prose-muted">
      <h1>Default Muted</h1>
      ${text}
    </div>
  </div>`
})

export const SM = () => ({
  template: `<div class="grid grid-cols-2">
    <div class="p-5 prose prose-sm">
      <h1>SM</h1>
      ${text}
    </div>
    <div class="p-5 prose prose-sm prose-muted">
      <h1>SM Muted</h1>
      ${text}
    </div>
  </div>`
})

export const LG = () => ({
  template: `<div class="grid grid-cols-2">
    <div class="p-5 prose prose-lg">
      <h1>LG</h1>
      ${text}
    </div>
    <div class="p-5 prose prose-lg prose-muted">
      <h1>LG Muted</h1>
      ${text}
    </div>
  </div>`
})

export const XL = () => ({
  template: `<div class="grid grid-cols-2">
    <div class="p-5 prose prose-xl">
      <h1>XL</h1>
      ${text}
    </div>
    <div class="p-5 prose prose-xl prose-muted">
      <h1>XL Muted</h1>
      ${text}
    </div>
  </div>`
})

export const TwoXL = () => ({
  template: `<div class="grid grid-cols-2">
    <div class="p-5 prose prose-2xl">
      <h1>2xl</h1>
      ${text}
    </div>
    <div class="p-5 prose prose-2xl prose-muted">
      <h1>2xl Muted</h1>
      ${text}
    </div>
  </div>`
})
