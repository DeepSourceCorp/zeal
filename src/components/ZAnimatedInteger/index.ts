import ZAnimatedInteger from './ZAnimatedInteger.vue'

export default ZAnimatedInteger
