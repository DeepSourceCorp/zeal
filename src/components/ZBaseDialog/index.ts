import ZBaseDialog from './ZBaseDialog.vue'

export default ZBaseDialog
