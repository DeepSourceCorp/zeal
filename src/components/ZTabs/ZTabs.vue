<template>
  <div class="z-tabs text-vanilla-400">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'ZTabs',
  props: {
    defaultActive: {
      type: Number,
      default: 0
    },
    active: {
      type: Number,
      default: 0
    }
  },
  data() {
    return { activeIndex: 0 }
  },
  computed: {
    getActiveIndex(): number {
      // Return the current active tab index
      return this.active >= 0 ? this.active : this.activeIndex
    }
  },
  mounted() {
    this.activeIndex = this.defaultActive
  },
  methods: {
    updateActiveIndex(index: number): void {
      const previous = this.activeIndex
      this.activeIndex = index
      this.$emit('onTabChange', { index, previous })
    }
  }
})
</script>
