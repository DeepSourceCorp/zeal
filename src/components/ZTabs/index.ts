import ZTabs from './ZTabs.vue'
import ZTab from './ZTab.vue'
import ZTabList from './ZTabList.vue'
import ZTabItems from './ZTabItems.vue'
import ZTabItem from './ZTabItem.vue'

export { ZTabs, ZTabList, ZTab, ZTabItems, ZTabItem }

export default ZTabs
