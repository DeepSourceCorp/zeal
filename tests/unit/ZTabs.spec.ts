import { ZTabs, ZTab, ZTabList, ZTabItems, ZTabItem } from '../../src/components/ZTabs'
import { mount } from '@vue/test-utils'

const BasicTab = {
  template: `<z-tabs>
              <z-tab-list>
                <z-tab>Overview</z-tab>
                <z-tab id="issues-tab">Issues</z-tab>
                <z-tab id="fixes-tab">Fixes</z-tab>
                <z-tab id="disabled-tab" :disabled="true">Metrics</z-tab>
              </z-tab-list>
              <z-tab-items>
                <z-tab-item>Overview of the code.</z-tab-item>
                <z-tab-item>There are the list of issues.</z-tab-item>
                <z-tab-item>There are the list of fixes.</z-tab-item>
                <z-tab-item>This are metrics.</z-tab-item>
              </z-tab-items>
            </z-tabs>`,
  components: {
    ZTabs,
    ZTab,
    ZTabList,
    ZTabItems,
    ZTabItem
  }
}

const TabsWithicon = {
  template: `<z-tabs>
              <z-tab-list>
                <z-tab icon="code">Overview</z-tab>
                <z-tab icon="code">Issues</z-tab>
                <z-tab icon="code">Metrics</z-tab>
              </z-tab-list>
              <z-tab-items>
                <z-tab-item>Overview of the code.</z-tab-item>
                <z-tab-item>There are the list of issues.</z-tab-item>
                <z-tab-item>This are metrics.</z-tab-item>
              </z-tab-items>
            </z-tabs>`,
  components: {
    ZTabs,
    ZTab,
    ZTabList,
    ZTabItems,
    ZTabItem
  }
}

describe('ZTabs', () => {
  it('renders a default tab component', () => {
    const wrapper = mount(BasicTab)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-tab').length).toEqual(4)
    expect(wrapper.findAll('.z-tab-item').length).toEqual(1)
  })

  it('renders tabs with icons', () => {
    const wrapper = mount(TabsWithicon)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-tab').length).toEqual(3)
    expect(wrapper.findAll('.z-tab > .z-icon').length).toEqual(3)
  })

  it('tabs switch on click', async () => {
    const wrapper = mount(BasicTab)

    wrapper.find('#issues-tab').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.z-tab-item').text()).toEqual('There are the list of issues.')

    wrapper.find('#fixes-tab').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.z-tab-item').text()).toEqual('There are the list of fixes.')

    // disabled click
    wrapper.find('#fixes-tab').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.z-tab-item').text()).toEqual('There are the list of fixes.')
  })
})
