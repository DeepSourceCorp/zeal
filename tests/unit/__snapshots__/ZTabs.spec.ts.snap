// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZTabs renders a default tab component 1`] = `
<div class="z-tabs text-vanilla-400">
  <div class="z-tab-list space-x-5 overflow-auto flex flex-nowrap"><button role="tab" aria-selected="true" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-100 border-juniper">
      <!----> Overview
    </button> <button role="tab" aria-selected="false" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-400 border-transparent" id="issues-tab">
      <!----> Issues
    </button> <button role="tab" aria-selected="false" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-400 border-transparent" id="fixes-tab">
      <!----> Fixes
    </button> <button role="tab" aria-selected="false" disabled="disabled" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-slate cursor-not-allowed border-transparent" id="disabled-tab">
      <!----> Metrics
    </button></div>
  <div role="tabpanel" tabindex="0" class="outline-none focus:outline-none z-tab-items">
    <div class="z-tab-pane">Overview of the code.</div>
  </div>
</div>
`;

exports[`ZTabs renders tabs with icons 1`] = `
<div class="z-tabs text-vanilla-400">
  <div class="z-tab-list space-x-5 overflow-auto flex flex-nowrap"><button role="tab" aria-selected="true" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-100 border-juniper"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--code text-vanilla-100">
        <polyline points="16 18 22 12 16 6"></polyline>
        <polyline points="8 6 2 12 8 18"></polyline>
      </svg> Overview</button> <button role="tab" aria-selected="false" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-400 border-transparent"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--code text-vanilla-400">
        <polyline points="16 18 22 12 16 6"></polyline>
        <polyline points="8 6 2 12 8 18"></polyline>
      </svg> Issues</button> <button role="tab" aria-selected="false" class="inline-flex items-end gap-2 pb-3 px-1 text-sm leading-none outline-none z-nav-item focus:outline-none border-b-2 z-tab-item text-vanilla-400 border-transparent"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--code text-vanilla-400">
        <polyline points="16 18 22 12 16 6"></polyline>
        <polyline points="8 6 2 12 8 18"></polyline>
      </svg> Metrics</button></div>
  <div role="tabpanel" tabindex="0" class="outline-none focus:outline-none z-tab-items">
    <div class="z-tab-pane">Overview of the code.</div>
  </div>
</div>
`;
